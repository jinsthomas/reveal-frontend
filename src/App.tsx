import { library } from '@fortawesome/fontawesome-svg-core';
import { faUser } from '@fortawesome/free-regular-svg-icons';
import React, { Component } from 'react';
import Header from './components/page/Header';

library.add(faUser);

import './App.css';

class App extends Component {
  public render() {
    return (
      <div className="container">
        <Header />
      </div>
    );
  }
}

export default App;
