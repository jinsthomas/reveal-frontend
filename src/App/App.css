tr.definitions td:first-child {
  padding-left: 0px;
}

h2.page-title {
  font-weight: 200;
}
