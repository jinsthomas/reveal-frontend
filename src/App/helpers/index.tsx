import { ComponentClass, FunctionComponent } from 'react';
import WithGATracker from '../../components/page/GoogleAnalytics/WithGATracker';
import { GA_ROUTE_COMPONENT } from '../../constants';

/**
 * Return component to be tracked by google analytics
 *
 * @param {ComponentClass | FunctionComponent} ComponentToTrack React component to be tracked
 * @param {string} trackingMethod Tracking approach
 */
export const trackedComponent = (
  ComponentToTrack: ComponentClass | FunctionComponent | ComponentClass<any, any>,
  trackingMethod: string = GA_ROUTE_COMPONENT
) => (trackingMethod === GA_ROUTE_COMPONENT ? ComponentToTrack : WithGATracker(ComponentToTrack));
