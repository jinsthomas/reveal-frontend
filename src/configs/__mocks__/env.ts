/** You can hard code variables that would have been
 * extracted from the environment here, and then mock
 * this file from your test file.
 * The goal is to have consistent test results
 */

export const SUPERSET_PLANS_SLICE = 0;
export type SUPERSET_PLANS_SLICE = typeof SUPERSET_PLANS_SLICE;

export const GISIDA_MAPBOX_TOKEN = 'hunter2';
export type GISIDA_MAPBOX_TOKEN = typeof GISIDA_MAPBOX_TOKEN;

export const GISIDA_ONADATA_API_TOKEN = 'hunter2';
export type GISIDA_ONADATA_API_TOKEN = typeof GISIDA_ONADATA_API_TOKEN;

export const DIGITAL_GLOBE_CONNECT_ID = 'hunter2';
export type DIGITAL_GLOBE_CONNECT_ID = typeof DIGITAL_GLOBE_CONNECT_ID;

export const OPENSRP_API_BASE_URL = 'https://test.smartregister.org/opensrp/rest/';
export type OPENSRP_API_BASE_URL = typeof OPENSRP_API_BASE_URL;

export const PLAN_UUID_NAMESPACE = '1bb3c217-af1c-4749-b3df-4f6108582e3c';
export type PLAN_UUID_NAMESPACE = typeof PLAN_UUID_NAMESPACE;

export const ACTION_UUID_NAMESPACE = '1bb3c217-af1c-4749-b3df-4f6108582e3c';
export type ACTION_UUID_NAMESPACE = typeof ACTION_UUID_NAMESPACE;
