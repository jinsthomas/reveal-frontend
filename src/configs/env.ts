/** This module handles settings taken from environment variables */

/** The website name */
export const WEBSITE_NAME = process.env.REACT_APP_WEBSITE_NAME || 'Reveal';
export type WEBSITE_NAME = typeof WEBSITE_NAME;

/** The domain name */
export const DOMAIN_NAME = process.env.REACT_APP_DOMAIN_NAME || 'http://localhost:3000';
export type DOMAIN_NAME = typeof DOMAIN_NAME;

/** Do you want to enable the IRS features? */
export const ENABLE_IRS = process.env.REACT_APP_ENABLE_IRS === 'true';
export type ENABLE_IRS = typeof ENABLE_IRS;

/** Do you want to enable the Focus Investigation features? */
export const ENABLE_FI = process.env.REACT_APP_ENABLE_FI === 'true';
export type ENABLE_FI = typeof ENABLE_FI;

/** Do you want to disable login protection? */
export const DISABLE_LOGIN_PROTECTION = process.env.REACT_APP_DISABLE_LOGIN_PROTECTION === 'true';
export type DISABLE_LOGIN_PROTECTION = typeof DISABLE_LOGIN_PROTECTION;

/** The Superset API base */
export const SUPERSET_API_BASE = process.env.REACT_APP_SUPERSET_API_BASE;
export type SUPERSET_API_BASE = typeof SUPERSET_API_BASE;

/** The Superset API endpoint */
export const SUPERSET_API_ENDPOINT = process.env.REACT_APP_SUPERSET_API_ENDPOINT;
export type SUPERSET_API_ENDPOINT = typeof SUPERSET_API_ENDPOINT;

/** OPENSRP Client ID, for oAuth2 */
export const OPENSRP_CLIENT_ID = process.env.REACT_APP_OPENSRP_CLIENT_ID;
export type OPENSRP_CLIENT_ID = typeof OPENSRP_CLIENT_ID;

/** Onadata Client ID, for oAuth2 */
export const ONADATA_CLIENT_ID = process.env.REACT_APP_ONADATA_CLIENT_ID;
export type ONADATA_CLIENT_ID = typeof ONADATA_CLIENT_ID;
