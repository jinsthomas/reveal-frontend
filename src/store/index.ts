import { combine } from '@onaio/redux-reducer-registry';
import reducerRegistry from '@onaio/redux-reducer-registry';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { applyMiddleware, compose, createStore, Reducer } from 'redux';
import thunk from 'redux-thunk';
import users, { reducerName as usersReducer } from './ducks/users';

/** Add redux devtools to Window */
declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

/** Declare history variable */
export const history = createBrowserHistory();

/** Declare type for reducer Registry */
export interface Registry {
  [key: string]: Reducer;
}

/** Function to create the store */
export function getStore(reducers: Registry) {
  /** Register each of the initial reducers */
  Object.keys(reducers).forEach(reducerName => {
    reducerRegistry.register(reducerName, reducers[reducerName]);
  });

  /** Combine reducers */
  const reducer = combine(reducerRegistry.getReducers());

  /** Add redux devtools to enhancers */
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  /** Create the store */
  return createStore(reducer, composeEnhancers(applyMiddleware(thunk, routerMiddleware(history))));
}

/** Router reducer */
const connectReducer = connectRouter(history);

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as any /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[usersReducer] = users;

/** The initial store for the reveal web app */
const store = getStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener(reducers => {
  store.replaceReducer(combine(reducers));
});

export default store;
