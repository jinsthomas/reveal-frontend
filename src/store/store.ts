import { combine } from '@onaio/redux-reducer-registry';
import reducerRegistry from '@onaio/redux-reducer-registry';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { applyMiddleware, createStore, Reducer } from 'redux';
import thunk from 'redux-thunk';
import users, { reducerName as usersReducer } from './ducks/users';

export const history = createBrowserHistory();

/** Declare type for reducer Registry */
export interface Registry {
  [key: string]: Reducer;
}

/** Function to create the store */
export function getStore(reducers: Registry) {
  /** Register each of the initial reducers */
  Object.keys(reducers).forEach(reducerName => {
    reducerRegistry.register(reducerName, reducers[reducerName]);
  });
  /** Combine reducers */
  const reducer = combine(reducerRegistry.getReducers());
  /** Create the store */
  return createStore(
    reducer,
    (window as any).__REDUX_DEVTOOLS_EXTENSION__ && (window as any).__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(thunk, routerMiddleware(history))
  );
}

/** Router reducer */
const connectReducer = connectRouter(history);
/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as any /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};
// /** Add users reducer to registry */
defaultReducers[usersReducer] = users;

/** The initial store for the reveal web app */
const store = getStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener(reducers => {
  store.replaceReducer(combine(reducers));
});

export default store;
