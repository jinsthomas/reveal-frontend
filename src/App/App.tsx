import { library } from '@fortawesome/fontawesome-svg-core';
import { faUser } from '@fortawesome/free-regular-svg-icons';
import { faMap } from '@fortawesome/free-solid-svg-icons';
import ConnectedPrivateRoute from '@onaio/connected-private-route';
import { ConnectedLogout, ConnectedOauthCallback, OauthLogin } from '@onaio/gatekeeper';
import React, { Component } from 'react';
import { Route, Switch } from 'react-router';
import { Col, Container, Row } from 'reactstrap';
import Loading from '../components/page/Loading';
import { DISABLE_LOGIN_PROTECTION } from '../configs/env';
import { providers } from '../configs/settings';
import {
  FI_HISTORICAL_URL,
  FI_SINGLE_MAP_URL,
  FI_SINGLE_URL,
  FI_URL,
  INTERVENTION_IRS_URL,
  LOGIN_URL,
  LOGOUT_URL,
} from '../constants';
import ConnectedHeader from '../containers/ConnectedHeader';
import ActiveFocusInvestigation from '../containers/pages/FocusInvestigation/active';
import HistoricalFocusInvestigation from '../containers/pages/FocusInvestigation/historical';
import SingleActiveFIMap from '../containers/pages/FocusInvestigation/map/active';
import SingleFI from '../containers/pages/FocusInvestigation/single';
import Home from '../containers/pages/Home/Home';
import IrsPlans from '../containers/pages/InterventionPlan/IRS';
import IrsDraftPlan from '../containers/pages/InterventionPlan/IRS/draft';
import IrsFinalizedPlan from '../containers/pages/InterventionPlan/IRS/finalized';
import IrsNewPlan from '../containers/pages/InterventionPlan/IRS/new';
import { oAuthUserInfoGetter } from '../helpers/utils';

library.add(faMap);
library.add(faUser);

import './App.css';

/** Main App component */
class App extends Component {
  public render() {
    return (
      <Container>
        <ConnectedHeader />
        <Row id="main-page-row">
          <Col>
            <Switch>
              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path="/"
                component={Home}
              />
              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={INTERVENTION_IRS_URL}
                component={IrsPlans}
              />

              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={`${INTERVENTION_IRS_URL}/new`}
                component={IrsNewPlan}
              />

              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={`${INTERVENTION_IRS_URL}/draft`} // todo - add :id
                component={IrsDraftPlan}
              />

              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={`${INTERVENTION_IRS_URL}/plan`} // todo - add :id
                component={IrsFinalizedPlan}
              />

              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={FI_URL}
                component={ActiveFocusInvestigation}
              />
              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={`${FI_SINGLE_URL}/:id`}
                component={SingleFI}
              />
              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={`${FI_SINGLE_MAP_URL}/:id/`}
                component={SingleActiveFIMap}
              />
              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={`${FI_SINGLE_MAP_URL}/:id/:goalId`}
                component={SingleActiveFIMap}
              />
              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={FI_HISTORICAL_URL}
                component={HistoricalFocusInvestigation}
              />
              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={`${FI_HISTORICAL_URL}/:id`}
                component={HistoricalFocusInvestigation}
              />
              {/* tslint:disable jsx-no-lambda */}
              <Route
                exact={true}
                path={LOGIN_URL}
                render={routeProps => <OauthLogin providers={providers} {...routeProps} />}
              />
              <Route
                exact={true}
                path="/oauth/callback/:id"
                render={routeProps => (
                  <ConnectedOauthCallback
                    LoadingComponent={Loading}
                    providers={providers}
                    oAuthUserInfoGetter={oAuthUserInfoGetter}
                    {...routeProps}
                  />
                )}
              />
              {/* tslint:enable jsx-no-lambda */}
              <ConnectedPrivateRoute
                disableLoginProtection={DISABLE_LOGIN_PROTECTION}
                exact={true}
                path={LOGOUT_URL}
                component={ConnectedLogout}
              />
            </Switch>
          </Col>
        </Row>
      </Container>
    );
  }
}

export default App;
