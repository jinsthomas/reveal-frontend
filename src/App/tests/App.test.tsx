import { updateExtraData } from '@onaio/session-reducer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { OPENSRP_LOGOUT_URL } from '../../configs/env';
import store from '../../store';
import App from '../App';

const history = createBrowserHistory();

describe('App', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <Router history={history}>
          <App />
        </Router>
      </Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders App correctly', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <App />
        </Router>
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
  it('logout url is set correctly Oath provider is opensrp', () => {
    window.open = jest.fn();
    store.dispatch(
      updateExtraData({
        oAuth2Data: {
          access_token: 'ec16dd21-b4f3-4ac1-8ede-e7c6f5d68c9e',
          expires_in: '3599',
          state: 'opensrp',
          token_type: 'bearer',
        },
        preferredName: 'Superset User',
        roles: ['Provider'],
        userName: 'superset-user',
      })
    );

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <App />
        </Router>
      </Provider>
    );

    history.push('/logout');
    expect(window.open).toBeCalledWith(OPENSRP_LOGOUT_URL);
  });

  it('logout url is set correctly when oath provider is not opensrp', () => {
    window.open = jest.fn();
    mount(
      <Provider store={store}>
        <Router history={history}>
          <App />
        </Router>
      </Provider>
    );

    history.push('/logout');
    expect(window.open).not.toBeCalled();
  });
});
