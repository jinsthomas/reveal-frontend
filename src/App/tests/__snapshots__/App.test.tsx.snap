// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <App>
    <Container
      tag="div"
    >
      <div
        className="container"
      >
        <Header>
          <div>
            <Navbar
              color="light"
              expand="md"
              light={true}
              tag="nav"
            >
              <nav
                className="navbar navbar-expand-md navbar-light bg-light"
              >
                <NavbarBrand
                  href="/"
                  tag="a"
                >
                  <a
                    className="navbar-brand"
                    href="/"
                  >
                    Reveal
                  </a>
                </NavbarBrand>
                <NavbarToggler
                  onClick={[Function]}
                  tag="button"
                  type="button"
                >
                  <button
                    className="navbar-toggler"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      className="navbar-toggler-icon"
                    />
                  </button>
                </NavbarToggler>
                <Collapse
                  appear={false}
                  enter={true}
                  exit={true}
                  in={false}
                  isOpen={false}
                  mountOnEnter={false}
                  navbar={true}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  tag="div"
                  timeout={350}
                  unmountOnExit={false}
                >
                  <Transition
                    appear={false}
                    enter={true}
                    exit={true}
                    in={false}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={350}
                    unmountOnExit={false}
                  >
                    <div
                      className="collapse navbar-collapse"
                      style={Object {}}
                    >
                      <Nav
                        className="mr-auto"
                        navbar={true}
                        tag="ul"
                        vertical={false}
                      >
                        <ul
                          className="mr-auto navbar-nav"
                        >
                          <NavItem
                            tag="li"
                          >
                            <li
                              className="nav-item"
                            >
                              <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav-link"
                                to="/"
                              >
                                <Route
                                  path="\\\\/"
                                >
                                  <Link
                                    aria-current="page"
                                    className="nav-link active"
                                    replace={false}
                                    style={Object {}}
                                    to="/"
                                  >
                                    <a
                                      aria-current="page"
                                      className="nav-link active"
                                      href="/"
                                      onClick={[Function]}
                                      style={Object {}}
                                    >
                                      Home
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </li>
                          </NavItem>
                          <NavItem
                            tag="li"
                          >
                            <li
                              className="nav-item"
                            >
                              <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav-link"
                                to="/irs"
                              >
                                <Route
                                  path="\\\\/irs"
                                >
                                  <Link
                                    aria-current={null}
                                    className="nav-link"
                                    replace={false}
                                    to="/irs"
                                  >
                                    <a
                                      aria-current={null}
                                      className="nav-link"
                                      href="/irs"
                                      onClick={[Function]}
                                    >
                                      IRS
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </li>
                          </NavItem>
                          <NavItem
                            tag="li"
                          >
                            <li
                              className="nav-item"
                            >
                              <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav-link"
                                to="/404"
                              >
                                <Route
                                  path="\\\\/404"
                                >
                                  <Link
                                    aria-current={null}
                                    className="nav-link"
                                    replace={false}
                                    to="/404"
                                  >
                                    <a
                                      aria-current={null}
                                      className="nav-link"
                                      href="/404"
                                      onClick={[Function]}
                                    >
                                      Focus Investigation
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </li>
                          </NavItem>
                          <NavItem
                            tag="li"
                          >
                            <li
                              className="nav-item"
                            >
                              <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav-link"
                                to="/404"
                              >
                                <Route
                                  path="\\\\/404"
                                >
                                  <Link
                                    aria-current={null}
                                    className="nav-link"
                                    replace={false}
                                    to="/404"
                                  >
                                    <a
                                      aria-current={null}
                                      className="nav-link"
                                      href="/404"
                                      onClick={[Function]}
                                    >
                                      Users
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </li>
                          </NavItem>
                          <NavItem
                            tag="li"
                          >
                            <li
                              className="nav-item"
                            >
                              <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="nav-link"
                                to="/404"
                              >
                                <Route
                                  path="\\\\/404"
                                >
                                  <Link
                                    aria-current={null}
                                    className="nav-link"
                                    replace={false}
                                    to="/404"
                                  >
                                    <a
                                      aria-current={null}
                                      className="nav-link"
                                      href="/404"
                                      onClick={[Function]}
                                    >
                                      About
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </li>
                          </NavItem>
                        </ul>
                      </Nav>
                      <Nav
                        className="ml-0"
                        navbar={true}
                        tag="ul"
                        vertical={false}
                      >
                        <ul
                          className="ml-0 navbar-nav"
                        >
                          <UncontrolledDropdown
                            inNavbar={true}
                            nav={true}
                          >
                            <Dropdown
                              active={false}
                              addonType={false}
                              direction="down"
                              inNavbar={true}
                              isOpen={false}
                              nav={true}
                              setActiveFromChild={false}
                              toggle={[Function]}
                            >
                              <Manager
                                className="dropdown nav-item"
                                onKeyDown={[Function]}
                                tag="li"
                              >
                                <li
                                  className="dropdown nav-item"
                                  onKeyDown={[Function]}
                                >
                                  <DropdownToggle
                                    aria-haspopup={true}
                                    caret={true}
                                    color="secondary"
                                    nav={true}
                                  >
                                    <a
                                      aria-expanded={false}
                                      aria-haspopup={true}
                                      className="dropdown-toggle nav-link"
                                      href="#"
                                      onClick={[Function]}
                                    >
                                      <FontAwesomeIcon
                                        border={false}
                                        className=""
                                        fixedWidth={false}
                                        flip={null}
                                        icon={
                                          Array [
                                            "far",
                                            "user",
                                          ]
                                        }
                                        inverse={false}
                                        listItem={false}
                                        mask={null}
                                        pull={null}
                                        pulse={false}
                                        rotation={null}
                                        size={null}
                                        spin={false}
                                        symbol={false}
                                        title=""
                                        transform={null}
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className="svg-inline--fa fa-user fa-w-14 "
                                          data-icon="user"
                                          data-prefix="far"
                                          focusable="false"
                                          role="img"
                                          style={Object {}}
                                          viewBox="0 0 448 512"
                                          xmlns="http://www.w3.org/2000/svg"
                                        >
                                          <path
                                            d="M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
                                            fill="currentColor"
                                            style={Object {}}
                                          />
                                        </svg>
                                      </FontAwesomeIcon>
                                       Roger
                                    </a>
                                  </DropdownToggle>
                                  <DropdownMenu
                                    flip={true}
                                    right={true}
                                    tag="div"
                                  >
                                    <div
                                      aria-hidden={true}
                                      className="dropdown-menu dropdown-menu-right"
                                      role="menu"
                                      tabIndex="-1"
                                    >
                                      <DropdownItem
                                        tag="button"
                                        toggle={true}
                                      >
                                        <button
                                          className="dropdown-item"
                                          onClick={[Function]}
                                          role="menuitem"
                                          tabIndex="0"
                                          type="button"
                                        >
                                          Sign Out
                                        </button>
                                      </DropdownItem>
                                    </div>
                                  </DropdownMenu>
                                </li>
                              </Manager>
                            </Dropdown>
                          </UncontrolledDropdown>
                        </ul>
                      </Nav>
                    </div>
                  </Transition>
                </Collapse>
              </nav>
            </Navbar>
          </div>
        </Header>
        <Row
          tag="div"
        >
          <div
            className="row"
          >
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <div
                className="col"
              >
                <Switch>
                  <Route
                    component={[Function]}
                    computedMatch={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                    exact={true}
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    path="/"
                  >
                    <Home
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                    >
                      <div
                        className="text-center"
                      >
                        <Row
                          className="welcome-box"
                          tag="div"
                        >
                          <div
                            className="welcome-box row"
                          >
                            <Col
                              tag="div"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <div
                                className="col"
                              >
                                <h3>
                                  Welcome to Reveal
                                </h3>
                                <p>
                                  Get started by selecting an intervention below
                                </p>
                              </div>
                            </Col>
                          </div>
                        </Row>
                        <Row
                          className="intervention-box"
                          tag="div"
                        >
                          <div
                            className="intervention-box row"
                          >
                            <Col
                              md="6"
                              tag="div"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <div
                                className="col-md-6"
                              >
                                <Link
                                  className="intervention-link"
                                  replace={false}
                                  to="/irs"
                                >
                                  <a
                                    className="intervention-link"
                                    href="/irs"
                                    onClick={[Function]}
                                  >
                                    <Button
                                      className="btn-intervention btn-lg btn-block btn-outline-dark"
                                      color="outline"
                                      tag="button"
                                    >
                                      <button
                                        aria-label={null}
                                        className="btn-intervention btn-lg btn-block btn-outline-dark btn btn-outline"
                                        onClick={[Function]}
                                      >
                                        IRS
                                      </button>
                                    </Button>
                                  </a>
                                </Link>
                              </div>
                            </Col>
                            <Col
                              md="6"
                              tag="div"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <div
                                className="col-md-6"
                              >
                                <Link
                                  className="intervention-link"
                                  replace={false}
                                  to="/404"
                                >
                                  <a
                                    className="intervention-link"
                                    href="/404"
                                    onClick={[Function]}
                                  >
                                    <Button
                                      className="btn-intervention btn-lg btn-block btn-outline-dark"
                                      color="outline"
                                      tag="button"
                                    >
                                      <button
                                        aria-label={null}
                                        className="btn-intervention btn-lg btn-block btn-outline-dark btn btn-outline"
                                        onClick={[Function]}
                                      >
                                        Focus Investigation
                                      </button>
                                    </Button>
                                  </a>
                                </Link>
                              </div>
                            </Col>
                          </div>
                        </Row>
                      </div>
                    </Home>
                  </Route>
                </Switch>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
  </App>
</Router>
`;
